Session4/ArrOperation.java

```diff
@@ -0,0 +1,185 @@
+					break;
+			}
+			if (CheckingStatus) {
+				clumps++;
+				FirstIndex = FirstIndex + LastIndex - 1;
+			}
+		}
+		System.out.println(clumps);
+		return clumps;
+	}
+
+	
+	static boolean checkadjecency(int[] inputarray ,int X) {
+		for(int count=0;count<inputarray.length-1;count++)
+		{
+			if(inputarray[count]==X && inputarray[count+1]==X) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	static int countOfX(int[] inputArray, int X) {
+		int countX = 0;
+		for (int count = 0; count < inputArray.length; count++) {
+			if (inputArray[count] == X)
+				countX++;
+		}
+		return countX;
+	}
+	
+	static int countOfY(int[] inputArray, int Y) {
+		int countY = 0;
+		for (int count = 0; count < inputArray.length; count++) {
+			if (inputArray[count] == Y)
+				countY++;
+		}
+		return countY;
+	}
+	
+	
+	/*
+	 *Return an array that contains exactly the same numbers as the input array,
+	 *but rearranged so that every X is immediately followed by a Y. 
+	 *Do not move X within array, but every other number may move.
+	 *
+	 */
+	static int[] fixXY(int[] inputarray ,int X,int Y) {
+		
+		//Assertion error case 1 ...
+		if (inputarray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+		//Assertion error case 2 ...
+		else if(checkadjecency(inputarray, X))
+			throw new AssertionError("Array should not contain adjacent X");
+		//Assertion error case 3 ...	
+		else if(countOfX(inputarray, X)!=countOfY(inputarray, Y))
+			throw new AssertionError("Array should contain equal Xs and Ys");
+		//Assertion error case 4 ...
+		else if(inputarray[inputarray.length-1] == X)
+			throw new AssertionError("Array should not contain last element as X");
+		/*
+		 * array list created to store the index of values of inputarray
+		 * which are equal to X(user input)
+		 */
+		ArrayList<Integer> index = new ArrayList<Integer>();
+		int []aux =new int[1]  ; 
+		for(int count=0;count<inputarray.length;count++)
+		{
+			if(count==0 && inputarray[count]==Y)
+				index.add(count);
+			else if(count>0 && inputarray[count]==Y && inputarray[count-1]!=X) {
+				index.add(count);
+			}
+		}	
+		for (int count = 0;count<inputarray.length; count++) {
+			if(inputarray[count]==X && inputarray[count+1]!=Y) {
+				aux[0]= inputarray[count+1];
+				inputarray[count+1]=inputarray[index.get(0)];
+				inputarray[index.get(0)]=aux[0];
+				index.remove(0);
+			}
+		}
+		return inputarray ;
+	}
+
+	
+	/*
+	 * Return the index if there is a place to split the input array so
+	 *  that the sum of the numbers on one side is equal to the sum 
+	 *  of the numbers on the other side else return -1 .
+	 */
+	static int split(int splitArray[]) {
+		int ReturnValue ;
+		
+		if (splitArray.length == 0)
+			throw new AssertionError("Array size cannot be zero");
+		
+		int sum = 0, leftSum = 0, rightSum = 0;
+		for (int i = 0; i < splitArray.length; i++) {
+			sum += splitArray[i];
+		}
+		if (sum % 2 != 0){
+			ReturnValue =-1;
+			return ReturnValue;
+	    }
+		int index;
```

Comment:
One method should have one or two return statements max.

Reply:
returnValue is the only return statement